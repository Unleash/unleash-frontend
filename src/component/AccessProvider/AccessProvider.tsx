import { FC } from "react";

import AccessContext from '../../contexts/AccessContext'
import { ADMIN } from "./permissions";

// TODO: Type up redux store
interface IAccessProvider {
    store: any;
}

interface IPermission {
  permission: string;
  project: string | null;
}

const AccessProvider: FC<IAccessProvider> = ({store, children}) => {
  const hasAccess = (permission: string, project: string) => {
    const permissions = store.getState().user.get('permissions') || [];

    const result = permissions.some((p: IPermission) => {
      if(p.permission === ADMIN) {
        return true
      }
      if(p.permission === permission && p.project === project) {
        return true;
      }
      return false;
    });

    return result;
  };

  const context = { hasAccess };

  return <AccessContext.Provider value={context}>{children}</AccessContext.Provider>
}

export default AccessProvider;