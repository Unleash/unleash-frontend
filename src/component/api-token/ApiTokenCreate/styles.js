import { makeStyles } from '@material-ui/styles';

export const useStyles = makeStyles({
    addApiKeyForm: {
        display: 'flex',
        flexDirection: 'column',
    },
});
