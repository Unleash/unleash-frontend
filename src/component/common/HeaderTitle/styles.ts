import makeStyles from '@mui/styles/makeStyles';

export const useStyles = makeStyles(theme => ({
    headerTitleContainer: {
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        position: 'relative',
    },
    headerTitle: {
        fontSize: theme.fontSizes.mainHeader,
        fontWeight: 'normal',
    },
    headerActions: {
        position: 'absolute',
        right: 0,
    },
}));
