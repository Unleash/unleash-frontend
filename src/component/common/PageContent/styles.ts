import { makeStyles } from 'tss-react/mui';

export const useStyles = makeStyles()(theme => ({
    headerContainer: {
        padding: theme.spacing(2, 4),
        borderBottom: `1px solid ${theme.palette.grey[100]}`,
        [theme.breakpoints.down('md')]: {
            padding: '1.5rem 1rem',
        },
    },
    bodyContainer: {
        padding: '2rem',
        [theme.breakpoints.down('md')]: {
            padding: '1rem',
        },
    },
    paddingDisabled: {
        padding: '0',
    },
    borderDisabled: {
        border: 'none',
    },
}));
