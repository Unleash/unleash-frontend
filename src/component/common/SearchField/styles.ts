import makeStyles from '@mui/styles/makeStyles';

export const useStyles = makeStyles(theme => ({
    container: {
        display: 'flex',
        alignItems: 'center',
        flexWrap: 'wrap',
        gap: '1rem',
    },
    search: {
        display: 'flex',
        alignItems: 'center',
        // @ts-expect-error
        backgroundColor: theme.palette.searchField.main,
        borderRadius: '25px',
        padding: '0.25rem 0.5rem',
        maxWidth: '450px',
        [theme.breakpoints.down('sm')]: {
            width: '100%',
        },
    },
    searchIcon: {
        marginRight: 8,
        color: theme.palette.grey[600],
    },
    inputRoot: {
        width: '100%',
    },
}));
