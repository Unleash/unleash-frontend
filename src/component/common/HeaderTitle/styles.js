import { makeStyles } from '@material-ui/styles';

export const useStyles = makeStyles(theme => ({
    headerTitleContainer: {
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        position: 'relative',
    },
    headerTitle: {
        fontSize: theme.fontSizes.mainHeader,
        fontWeight: 500,
    },
    headerActions: {
        position: 'absolute',
        right: 0,
    },
}));
