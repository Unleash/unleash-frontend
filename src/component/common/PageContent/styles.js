import { makeStyles } from '@material-ui/styles';

export const useStyles = makeStyles(theme => ({
    headerContainer: {
        padding: theme.padding.pageContent.header,
        borderBottom: theme.borders.default,
    },
    bodyContainer: {
        padding: theme.padding.pageContent.body,
    },
}));
