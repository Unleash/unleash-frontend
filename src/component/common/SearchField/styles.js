import { makeStyles } from "@material-ui/styles";

export const useStyles = makeStyles(theme => ({
    search: {
        display: "flex",
        alignItems: "center",
        backgroundColor: "#fff",
        borderRadius: "3px",
        padding: "0.25rem 0.5rem",
        maxWidth: "450px"
    },
    searchIcon: {
        marginRight: "8px"
    },
    inputRoot: {
        width: "100%"
    }
}));
