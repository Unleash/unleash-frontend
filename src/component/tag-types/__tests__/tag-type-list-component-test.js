import React from 'react';

import TagTypesList from '../TagTypeList';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from '@material-ui/styles';
import theme from '../../../themes/main-theme';
import { createFakeStore } from '../../../accessStoreFake';
import AccessProvider from '../../providers/AccessProvider/AccessProvider';

import {
    ADMIN,
    CREATE_TAG_TYPE,
    UPDATE_TAG_TYPE,
    DELETE_TAG_TYPE,
} from '../../providers/AccessProvider/permissions';
import UIProvider from '../../providers/UIProvider/UIProvider';

test('renders an empty list correctly', () => {
    const tree = renderer.create(
        <MemoryRouter>
            <ThemeProvider theme={theme}>
                <UIProvider>
                    <AccessProvider
                        store={createFakeStore([{ permission: ADMIN }])}
                    >
                        <TagTypesList
                            tagTypes={[]}
                            fetchTagTypes={jest.fn()}
                            removeTagType={jest.fn()}
                            history={{}}
                        />
                    </AccessProvider>
                </UIProvider>
            </ThemeProvider>
        </MemoryRouter>
    );
    expect(tree).toMatchSnapshot();
});

test('renders a list with elements correctly', () => {
    const tree = renderer.create(
        <ThemeProvider theme={theme}>
            <MemoryRouter>
                <UIProvider>
                    <AccessProvider
                        store={createFakeStore([
                            { permission: CREATE_TAG_TYPE },
                            { permission: UPDATE_TAG_TYPE },
                            { permission: DELETE_TAG_TYPE },
                        ])}
                    >
                        <TagTypesList
                            tagTypes={[
                                {
                                    name: 'simple',
                                    description: 'Some simple description',
                                    icon: '#',
                                },
                            ]}
                            fetchTagTypes={jest.fn()}
                            removeTagType={jest.fn()}
                            history={{}}
                        />
                    </AccessProvider>
                </UIProvider>
            </MemoryRouter>
        </ThemeProvider>
    );
    expect(tree).toMatchSnapshot();
});
