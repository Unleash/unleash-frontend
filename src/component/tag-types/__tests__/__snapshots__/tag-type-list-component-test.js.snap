// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list with elements correctly 1`] = `
<div
  className="MuiPaper-root fullwidth MuiPaper-elevation1 MuiPaper-rounded"
  shadow={0}
  style={
    Object {
      "overflow": "visible",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "borderBottom": "1px solid #f9f9f9",
        "display": "flex",
        "marginBottom": "10px",
        "padding": "16px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": "2",
        }
      }
    >
      <h6
        className="headerTitle"
      >
        Tag Types
      </h6>
    </div>
    <div
      style={
        Object {
          "flex": "1",
          "textAlign": "right",
        }
      }
    >
      <button
        aria-describedby={null}
        aria-label="add tag type"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Add tag type"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.63 5.84C17.27 5.33 16.67 5 16 5L5 5.01C3.9 5.01 3 5.9 3 7v10c0 1.1.9 1.99 2 1.99L16 19c.67 0 1.27-.33 1.63-.84L22 12l-4.37-6.16z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root MuiListItemText-multiline"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          <a
            href="/tag-types/edit/simple"
            onClick={[Function]}
          >
            <strong>
              simple
            </strong>
          </a>
        </span>
        <p
          className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          Some simple description
        </p>
      </div>
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </li>
  </ul>
</div>
`;

exports[`renders an empty list correctly 1`] = `
<div
  className="MuiPaper-root fullwidth MuiPaper-elevation1 MuiPaper-rounded"
  shadow={0}
  style={
    Object {
      "overflow": "visible",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "borderBottom": "1px solid #f9f9f9",
        "display": "flex",
        "marginBottom": "10px",
        "padding": "16px",
      }
    }
  >
    <div
      style={
        Object {
          "flex": "2",
        }
      }
    >
      <h6
        className="headerTitle"
      >
        Tag Types
      </h6>
    </div>
    <div
      style={
        Object {
          "flex": "1",
          "textAlign": "right",
        }
      }
    >
      <button
        aria-describedby={null}
        aria-label="add tag type"
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Add tag type"
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      No entries
    </li>
  </ul>
</div>
`;
